### DISCUSSION TITLE
Toggle GPU Hardware Acceleration

### DISCUSSION THREAD

**Discussion by Mantas-2155X** [mullvad/mullvadvpn-app#9053] [Feature requests & Ideas]
## Issue report

Operating system: Linux (Ubuntu 22.04)

App version: 2022.2

## Issue description

Ability to toggle GPU Hardware acceleration for the GUI application.

I have noticed that the mullvad gui app uses more than 200 MB of VRAM when sitting idle in the background. Since the GUI app is only interacted with quite rarely by the user to switch servers, I would suggest having an option to disable GPU acceleration (`--disable-gpu`) to save VRAM for lower end systems.

As a temporary workaround adding the above parameter to the `.desktop` file drops the VRAM usage to 4 MB and no slowdown is experienced (the animations are very simple after all).

**Comment by faern** [comment]
Disabling GPU acceleration definitely has performance impacts on some systems. But I agree that we should use way less VRAM on idle! We'll see what we can do.

**Comment by DPTJKKVH** [comment]
Hey @Mantas-2155X , can you please tell me on what position I should add the `--disable-gpu` to as shown in my picture? (Forget the 1, I realized that this is actually just a space in the path. Also I know that I have to add a space, don't worry.)
![unknown](https://user-images.githubusercontent.com/7238653/189038956-378b4f81-d9b0-40d3-a613-9746860068d1.png)

**Comment by raksooo** [comment]
@DPTJKKVH It should go at 3. The resulting line should be
```
Exec="/opt/Mullvad VPN/mullvad-vpn" --disable-gpu %U
```

**Comment by DPTJKKVH** [comment]
Thanks @raksooo ! Much appreciated. Really does make no difference for performance but saves a lot of VRAM!

**Comment by novacrazy** [comment]
I'd like to bump this. It's silly that Mullvad would consume 50-200MB of VRAM on just a taskbar icon 99.9% of the time. I'm about to update to the latest version and am unsure if I'll have to search through the registry to modify things again. 

I have never observed any performance decrease, though I do have a powerful CPU. If anything, it's faster. I use the same `--disable-gpu` trick on VS Code and other things with no noticeable change in performance, too. Though it is notable that Electron will create extra processes to do CPU rendering, and those will consume about the same amount of regular RAM as was previously consumed as VRAM. 

EDIT: Finally restarted after a few weeks. The `Mullvad VPN.exe` that autostarts was indeed reset, so I had to edit that again to save that 50MB VRAM.

**Comment by novacrazy** [comment]
Is there any chance we'll ever have this option?

**Comment by hulthe** [comment]
@novacrazy Maybe someday, but we have no plans to add this button right now.
